src/Symfony/Component/DependencyInjection/ServiceLocator.php

```diff
@@ -53,13 +52,12 @@ public function get($id)
             throw new ServiceCircularReferenceException($id, array($id, $id));
```

Comment:
Why should we remove the safe guard? Symfony is used into the wild, it needs to handle properly any situation. I see no reason to trade a few lines vs a fatal error, however rare it is.